ui/components/app/assets/asset-list/sort-control/sort-control.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState, ReactNode } from 'react';
+import classnames from 'classnames';
+import { Box } from '../../../../component-library';
+import { TokenWithBalance } from '../asset-list';
+import { SortOrder, SortingCallbacksT, sortAssets } from '../../util/sort';
+import {
+  BackgroundColor,
+  BorderRadius,
+} from '../../../../../helpers/constants/design-system';
+
+// intentionally used generic naming convention for styled selectable list item
+// inspired from ui/components/multichain/network-list-item
+// should probably be broken out into component library
+type SelectableListItemProps = {
+  isSelected: boolean;
+  onClick?: React.MouseEventHandler<HTMLSpanElement>;
+  children: ReactNode;
+};
+
+export const SelectableListItem = ({
+  isSelected,
+  onClick,
+  children,
+}: SelectableListItemProps) => {
+  return (
+    <Box className="selectable-list-item-wrapper">
+      <Box
+        className={classnames('selectable-list-item', {
+          'selectable-list-item--selected': isSelected,
+        })}
+        onClick={onClick}
+      >
+        {children}
+      </Box>
+      {isSelected && (
+        <Box
+          className="selectable-list-item__selected-indicator"
+          borderRadius={BorderRadius.pill}
+          backgroundColor={BackgroundColor.primaryDefault}
+        />
+      )}
+    </Box>
+  );
+};
```

Comment:
@georgewrmarshall I'm tagging you here because we talked about potentially breaking this out into the component library. Not sure if we want to do that here or in a different PR.

Reply:
A different PR would be great but I don't want it to halt progress, I could help you get it in a state tomorrow if that works!